### ISSUE TITLE
Seed with custom column type of string fails with boolean like seed values

### ISSUE THREAD

**Issue by ianterrell** [fishtown-analytics/dbt#999]
### Issue description

Seeding fails to treat a boolean-like string as a varchar even when specified.

I have the following seed file:

```csv
true_is_correct,is_correct,choice_body
true,true,"True"
true,false,"False"
false,true,"False"
false,false,"True"
```

With the following config in `dbt_project.yml`:

```yaml
seeds:
  project_name:
    enabled: true
    seed_file_name:
      column_types:
        choice_body: varchar(5)
```

### Results

`dbt seed` errors on this file with:

```
Database Error in seed seed_file_name (data/seed_file_name.csv)
  column "choice_body" is of type character varying but expression is of type boolean
  HINT:  You will need to rewrite or cast the expression.
```

I expect this to seed properly.

### System information

- dbt 0.11.0
- Mac OS X 10.13.6 (17G65)
- Python 2.7.10
- Redshift

### Steps to reproduce

Sample seed file and config above.

**Comment by drewbanin** [comment]
Thanks for the report @ianterrell - we'll take a look!

**Comment by borisuvarov** [comment]
@drewbanin Hi Drew, we're experiencing the same issue: dbt implicitly processes values from the seed files before it loads it to a seed table, so even if data type of the column is specified via `column_types` (e.g. VARCHAR(50)), the values will be changed (e.g., leading zero is going to be stripped, if dbt thinks that values in that column are integers, not strings). Are there any plans to to fix this? Thanks in advance!

**Comment by drewbanin** [comment]
hey @borisuvarov - thanks for the update. dbt doesn't do any of this type inference itself. Instead, the work is outsourced to a library called [agate](https://github.com/wireservice/agate).

I 100% agree with you that dbt's behavior in these situations is incorrect and suboptimal, but I haven't yet been able to figure out a good way to make this work.

One option is to use `column_types` to specify both the database type, and the python type. That could look like:
```
column_types:
  some_field:
    sql: varchar(100)
    type: string
```

^ I don't at all like that syntax, but I _think_ we need both of these pieces of information! I would like to get this working at some point, but we don't plan on prioritizing a fix for this in the near future.  If you have good ideas for how this could work, I'd love to hear them!

**Comment by borisuvarov** [comment]
Thanks @drewbanin! 

Adding a `type` should help in most cases. But it's also worth to try to figure out how to (optionally) leverage native COPY (Redshift), COPY INTO (Snowflake), etc. DML commands, which loads a CSV file without any transformations of values in Python.

**Comment by drewbanin** [comment]
Going to close this - it may still be an issue, but we're largely at the behest of Agate for seed types. Happy to discuss / reopen if anyone feels strongly about it!

**Comment by beckjake** [comment]
Fixed in #1920
